provider/k8s.go

```diff
@@ -321,16 +376,52 @@ func (store genericBlobStore) Serve(key string) (Iterator, error) {
 	switch fileType := keyParts[len(keyParts)-1]; fileType {
 	case "csv":
 		return csvIteratorFromReader(r)
+	case "parquet":
+		return parquetIteratorFromReader(r)
 	default:
 		return nil, fmt.Errorf("unsupported file type")
 	}
 
 }
 
+type ParquetIterator struct {
+	rows  []interface{}
+	index int64
+}
+
+func (p *ParquetIterator) Next() (interface{}, error) {
+	if p.index+1 == int64(len(p.rows)) {
+		return nil, fmt.Errorf("end of iterator")
+	}
+	p.index += 1
+	return p.rows[p.index], nil
+}
+
+func parquetIteratorFromReader(r io.Reader) (Iterator, error) {
+	buff := bytes.NewBuffer([]byte{})
+	size, err := io.Copy(buff, r)
+	if err != nil {
+		return nil, err
+	}
+	file := bytes.NewReader(buff.Bytes())
+	rows, err := parquet.Read[any](file, size)
+	if err != nil {
+		return nil, err
+	}
+	return &ParquetIterator{
+		rows:  rows,
+		index: int64(0),
+	}, nil
+}
+
 func (store genericBlobStore) Exists(key string) (bool, error) {
 	return store.bucket.Exists(ctx, key)
 }
 
+func (store genericBlobStore) Delete(key string) error {
```

Comment:
how is this functionality being used? Should we give delete capabilities?

Reply:
it's used for deletematerialization